clippy_lints/src/format.rs

```diff
@@ -92,7 +92,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for UselessFormat {
     }
 }
 
-fn span_useless_format<'a, 'tcx: 'a, T: LintContext<'tcx>>(cx: &'a T, span: Span, help: &str, mut sugg: String) {
+fn span_useless_format<T: LintContext>(cx: &T, span: Span, help: &str, mut sugg: String) {
```

Comment:
cc @oli-obk `T` should be named `C` IMO, or maybe `cx: &impl LintContext` can be used (and, well, `LintContext` should probably be just `LintCx`).